feature/preview/src/main/java/com/google/jetpackcamera/feature/preview/PreviewScreen.kt

```diff
@@ -92,11 +89,9 @@ fun PreviewScreen(
 
     val lifecycleOwner = LocalLifecycleOwner.current
 
-    val deferredSurfaceProvider = remember { CompletableDeferred<SurfaceProvider>() }
-    val onSurfaceProviderReady: (SurfaceProvider) -> Unit = {
-        Log.d(TAG, "onSurfaceProviderReady")
-        deferredSurfaceProvider.complete(it)
-    }
+    val (surfaceProvider, onSurfaceProviderReady) =
```

Comment:
Consider renaming this to:
`val surfaceProvider by remember { mutableStateOf<SurfaceProvider?>(null) }`

And then when you call the CameraPreview composable, you use:
```
CameraPreview(
                        ..
                        onSurfaceProviderReady = { surfaceProvider = it },
                        ..
                    )
```

Reply:
If you want to use it like this, I'd recommend using the name `setSurfaceProvider`. Using `onSurfaceProviderReady` refers to behavior which is not part of the mutable state itself, rather the way you're using it.